include/trainer_info.h

```diff
@@ -19,39 +19,291 @@ typedef struct TrainerInfo {
+
+/**
+ * @brief Set the flag denoting possession of a given badge.
+ * 
+ * @param info      The trainer info.
+ * @param badgeNum  Number of the badge to set.
+ */
+void TrainerInfo_SetBadge(TrainerInfo *info, int badgeNum);
+
+/**
+ * @brief Get the number of badges owned by the trainer.
+ * 
+ * @param info  The trainer info.
+ * @return Number of badges acquired.
+ */
+int TrainerInfo_BadgeCount(const TrainerInfo *info);
+
+/**
+ * @brief Get the trainer's current money.
+ * 
+ * @param info  The trainer info.
+ * @return How much money the trainer has.
+ */
+u32 TrainerInfo_Money(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's current money.
+ * 
+ * @param info  The trainer info.
+ * @param money How much money the trainer should have.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_SetMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Get the trainer's appearance in the UNion Room.
+ * 
+ * @param info
+ * @return The trainer's Union Room apperance.
+ */
+u8 TrainerInfo_Appearance(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's appearance in the Union Room.
+ * 
+ * @param info          The trainer info.
+ * @param appearance    New appearance setting.
+ */
+void TrainerInfo_SetAppearance(TrainerInfo *info, u8 appearance);
+
+/**
+ * @brief Give the trainer a specified amount of money.
+ * 
+ * @param info  The trainer info.
+ * @param money The amount of money to give the trainer.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_GiveMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Take a specified amount of money from the trainer.
+ * 
+ * @param info  The trainer info.
+ * @param money The amount of money to take from the trainer.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_TakeMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Get the trainer's game code.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's game code.
+ */
+u8 TrainerInfo_GameCode(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's game code.
+ * 
+ * @param info      The trainer info.
+ * @param gameCode  New game code.
+ */
+void TrainerInfo_SetGameCode(TrainerInfo *info, u8 gameCode);
+
+/**
+ * @brief Get the game code for interfacing with Diamond and Pearl.
+ * 
+ * @return 0
+ */
 u8 TrainerInfo_DPGameCode(void);
-u8 TrainerInfo_RegionCode(const TrainerInfo * param0);
-void TrainerInfo_SetRegionCode(TrainerInfo * param0, u8 param1);
-void TrainerInfo_SetGameCleared(TrainerInfo * param0);
-int TrainerInfo_GameCleared(TrainerInfo * param0);
-void TrainerInfo_GiveNationalDex(TrainerInfo * param0);
-int TrainerInfo_HasNationalDex(TrainerInfo * param0);
-BOOL TrainerInfo_Equals(const TrainerInfo * param0, const TrainerInfo * param1);
+
+/**
+ * @brief Get the trainer's region code.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's region code.
+ */
+u8 TrainerInfo_RegionCode(const TrainerInfo *info);
```

Comment:
We should call this `TrainerInfo_GetRegionCode`, for consistency with `TrainerInfo_SetRegionCode`

Reply:
We've been moving in the direction of removing `Get` from accessors, as I recall. Names like this more closely mirror the underlying arrow notation (`info->regionCode`). Ideally, I'd purge all of these and replace them with direct arrow notation, but, *c'est la vie*.